src/lib/forms/rjsf/widgets/RJSFESSelector.js

```diff
@@ -0,0 +1,164 @@
+      value: value,
+      error: null,
+      options: [],
+    };
+  }
+
+  async componentDidMount() {
+    const { value } = this.props;
+    if (value) {
+      this.fetchValue(value);
+    }
+  }
+
+  componentWillUnmount() {
+    this.cancellableFetchData && this.cancellableFetchData.cancel();
+  }
+
+  async fetchValue(value) {
+    const { options } = this.props;
+    const { apiGetByValue, apiGetByValueResponseSerializer } = options;
+    this.cancellableFetchData = withCancel(apiGetByValue(value));
+    try {
+      this.setState({
+        isLoading: true,
+      });
+
+      const response = await this.cancellableFetchData.promise;
+      const singleOption = apiGetByValueResponseSerializer(response.data);
+
+      this.setState({
+        isLoading: false,
+        value: singleOption.value,
+        options: [singleOption],
+      });
+    } catch (error) {
+      if (error !== 'UNMOUNTED') {
+        this.setState({
+          isLoading: false,
+          error: 'Error loading results.',
+        });
+      }
+    }
+  }
+
+  handleSearchChange = async (e, { searchQuery }) => {
+    const { options, required } = this.props;
+    const { apiQuery, apiQueryResponseSerializer } = options;
+    if (searchQuery === '') {
+      return;
+    }
+
+    this.cancellableFetchData = withCancel(apiQuery(searchQuery));
+    try {
+      this.setState({
+        isLoading: true,
+      });
+
+      const response = await this.cancellableFetchData.promise;
+      const results = response.data.hits.map((hit) =>
+        apiQueryResponseSerializer(hit)
+      );
+
+      if (!required) {
+        // add an empty element at the beginning so that if it is not required, you can select nothing
+        results.unshift({ key: '', value: undefined, text: '' });
+      }
+
+      this.setState({
+        isLoading: false,
+        options: results,
+        error: null,
+      });
+    } catch (error) {
+      if (error !== 'UNMOUNTED') {
+        this.setState({
+          isLoading: false,
+          error: 'Error loading results.',
+        });
+      }
+    }
+  };
+
+  handleChange = (e, { value }) => {
+    const { onChange } = this.props;
+    this.setState({ value: value });
+    onChange(value);
+  };
+
+  render() {
+    const {
+      autofocus,
+      debounceDelay,
+      label,
+      placeholder,
+      readonly,
+      required,
+    } = this.props;
+    const { options, isLoading, value, error } = this.state;
+    return (
+      <Form.Select
+        fluid
+        selection
+        search
+        options={options}
+        label={label}
+        value={value}
+        required={required}
+        autoFocus={autofocus}
+        readOnly={readonly}
+        placeholder={placeholder}
+        onChange={this.handleChange}
+        onSearchChange={_debounce(this.handleSearchChange, debounceDelay)}
+        disabled={isLoading}
+        loading={isLoading}
+        noResultsMessage={error ? error : 'No results found'}
```

Comment:
Minor: Could we add and pass an `icon` prop? For some fields it may be useful, like for the patron, where we can add the search icon.
```diff
+        icon={icon}
```
![image](https://user-images.githubusercontent.com/15194802/107957019-4d769680-6fa0-11eb-839f-54c4b2f89daa.png)